packages/client/test/cli/cli.spec.ts

```diff
@@ -24,25 +51,145 @@ describe('[CLI]', () => {
+    const cliArgs = ['--rpcEngine', '--rpcEnginePort=8552', '--dev=poa']
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('http://')) {
+          // if http endpoint startup message detected, call http endpoint with RPC method
+          assert.ok(message.includes('8552'), 'read from HTTP RPC')
+          child.kill()
+          resolve(undefined)
+        }
+        if (message.toLowerCase().includes('error')) {
+          child.kill(9)
+          assert.fail(`client encountered error: ${message}`)
+        }
+      })
+      child.stderr.on('data', (data) => {
+        const message: string = data.toString()
+        assert.fail(`stderr: ${message}`)
+      })
+      child.on('close', (code) => {
+        if (typeof code === 'number' && code > 0) {
+          assert.fail(`child process exited with code ${code}`)
+        }
+      })
+    })
+  }, 18000)
+  it('should start engine rpc on custom address', async () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = ['--rpcEngine', '--rpcEngineAddr="0.0.0.0"', '--dev=poa']
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('http://')) {
+          // if http endpoint startup message detected, call http endpoint with RPC method
+          assert.ok(message.includes('0.0.0.0'), 'read from HTTP RPC')
+          child.kill()
+          resolve(undefined)
+        }
+        if (message.toLowerCase().includes('error')) {
+          child.kill(9)
+          assert.fail(`client encountered error: ${message}`)
+        }
+      })
+      child.stderr.on('data', (data) => {
+        const message: string = data.toString()
+        assert.fail(`stderr: ${message}`)
+      })
+      child.on('close', (code) => {
+        if (typeof code === 'number' && code > 0) {
+          assert.fail(`child process exited with code ${code}`)
+        }
+      })
+    })
+  }, 18000)
+  it('should start engine websocket on custom address and port', async () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = [
+      '--ws',
+      '--rpcEngine',
+      '--wsEnginePort=8552',
+      '--wsEngineAddr="0.0.0.0"',
+      '--dev=poa',
+    ]
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('ws://') && message.includes('engine')) {
+          // if http endpoint startup message detected, call http endpoint with RPC method
+          assert.ok(message.includes('ws://0.0.0.0:8552'), 'read from HTTP RPC')
```

Comment:
Change the success message to `read from WS RPC on custom address` and add an actual call to the endpoint (and verify you get the invalid credentials error)